File: ruby/sinatra/app/controllers/users/sessions_controller.rb
module Users
  class SessionsController < ApplicationController
    def create
      user = User.find_by(email: user_params[:email])

      if user.present? && user.password == user_params[:password]
        response.status = 200
        response.headers["Authorization"] = user.jwt
        user.to_json
      else
        response.status = 401
        { errors: ["Invalid Email or Password"] }.to_json
      end
    end

    def destroy
      current_user.edit_jti
      current_user.save
      response.status = 200
    end

    private

    def user_params
      params[:user]&.slice(:email, :password) || {}
    end
  end
end

File: node/express/src/controllers/users/sessions.controller.ts
import type { RequestHandler } from "express"
import bcrypt from "bcrypt"
import prisma from "src/models/user.model.ts"

export const create: RequestHandler = async (req, res) => {
  const { user: { email, password } } = req.body as { user: { email?: string, password?: string } }

  if (email && password) {
    const user = await prisma.user.findUnique({ where: { email } })
    if (user && await bcrypt.compare(password, user.password)) {
      return res.status(200).set("Authorization", user.jwt).json(user.toJSON())
    }
  }

  return res.status(401).json({ errors: { user: "Invalid Email or Password" } })
}

export const destroy: RequestHandler = async (req, res) => {
  await req.currentUser.editJTI()
  res.status(200).send()
}
